sdks/python/apache_beam/ml/rag/ingestion/alloydb.py

```diff
@@ -0,0 +1,494 @@
+        ...     jdbc_url='jdbc:postgresql://localhost:5432/mydb',
+        ...     username='user',
+        ...     password='pass',
+        ...     connection_properties={'ssl': 'true'},
+        ...     max_connections=10
+        ... )
+    """
+  jdbc_url: str
+  username: str
+  password: str
+  connection_properties: Optional[Dict[str, str]] = None
+  connection_init_sqls: Optional[List[str]] = None
+  autosharding: Optional[bool] = None
+  max_connections: Optional[int] = None
+  write_batch_size: Optional[int] = None
+
+
+@dataclass
+class ConflictResolution:
+  """Specification for how to handle conflicts during insert.
+
+    Configures conflict handling behavior when inserting records that may
+    violate unique constraints.
+
+    Attributes:
+        on_conflict_fields: Field(s) that determine uniqueness. Can be a single
+            field name or list of field names for composite constraints.
+        action: How to handle conflicts - either "UPDATE" or "IGNORE".
+            UPDATE: Updates existing record with new values.
+            IGNORE: Skips conflicting records.
+        update_fields: Optional list of fields to update on conflict. If None,
+            all non-conflict fields are updated.
+        
+    Examples:
+        Simple primary key:
+        >>> ConflictResolution("id")
+        
+        Composite key with specific update fields:
+        >>> ConflictResolution(
+        ...     on_conflict_fields=["source", "timestamp"],
+        ...     action="UPDATE",
+        ...     update_fields=["embedding", "content"]
+        ... )
+        
+        Ignore conflicts:
+        >>> ConflictResolution(
+        ...     on_conflict_fields="id",
+        ...     action="IGNORE"
+        ... )
+    """
+  on_conflict_fields: Union[str, List[str]]
+  action: Literal["UPDATE", "IGNORE"] = "UPDATE"
+  update_fields: Optional[List[str]] = None
+
+  def get_conflict_clause(self) -> str:
+    """Get conflict clause with update fields."""
+    fields = [self.on_conflict_fields] \
+      if isinstance(self.on_conflict_fields, str) \
+        else self.on_conflict_fields
+
+    if self.action == "IGNORE":
+      return f"ON CONFLICT ({', '.join(fields)}) DO NOTHING"
+
+    # update_fields should be set by query builder before this is called
+    assert self.update_fields is not None, \
+      "update_fields must be set before generating conflict clause"
+    updates = [f"{field} = EXCLUDED.{field}" for field in self.update_fields]
+    return f"ON CONFLICT " \
+      f"({', '.join(fields)}) DO UPDATE SET {', '.join(updates)}"
+
+
+@dataclass
+class ColumnSpec:
+  """Specification for mapping Chunk fields to SQL columns for insertion.
+    
+    Defines how to extract and format values from Chunks into database columns,
+    including type conversion and SQL type casting.
+
+    Attributes:
+        name: The column name in the database table.
+        python_type: Python type for the column value (used in NamedTuple field)
```

Comment:
Is there a set of supported types here? Is it basically anything that Alloy supports/is there a doc we could link to?